File: DownloadFiles/TGNet/Chapter06/VB/Exercise6-4/BugReporter/BugReporter/BugReporterForm.vb
Public Class BugReporterForm

  Private Sub btnSubmitBug_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSubmitBug.Click
    'Create a new object from the ReportToFile class
    Dim objReport As New ReportToFile

    'Create string variables and get the data from the User
    objReport.Tester = txtTester.Text
    objReport.ApplicationName = txtAppName.Text
    objReport.BuildNumber = txtBuildNumber.Text
    objReport.DateReported = txtDateReported.Text
    objReport.Description = txtDescription.Text
    objReport.FileName = txtFileName.Text

    'Call the FormatData method
    objReport.FormatData()

    'Call the Save method
    If objReport.Save() = True Then
      MessageBox.Show("Bug reported!")
      txtDescription.Text = ""
    Else
      MessageBox.Show("There was an error!")
    End If
  End Sub

End Class

File: DownloadFiles/TGNet/Chapter06/C#/Exercise6-4/BugReporter/BugReporter/BugReporterForm.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

#endregion

namespace BugReporter
{
  partial class BugReporterForm : Form
  {
    public BugReporterForm()
    {
      InitializeComponent();
    }

    ReportToFile objReport = new ReportToFile();

    private void btnSubmitBug_Click(object sender, EventArgs e)
    {
      //---- Get Data Section ----
      objReport.Tester = txtTester.Text;
      objReport.ApplicationName = txtAppName.Text;
      objReport.BuildNumber = txtBuildNumber.Text;
      objReport.DateReported = txtDateReported.Text;
      objReport.Description = txtDescription.Text;
      objReport.FileName = txtFileName.Text;

      objReport.FormatData();
    
    // Call the Save method
  if (objReport.Save() == true) 
  {
    MessageBox.Show("Bug recorded");
    txtDescription.Text = "";
  }
  else
  {
    MessageBox.Show("There was an error");
  }
    
    }//end of btnSubmitBug_Click()
    

  }//end of BugReporterForm class

}